### PR TITLE
fix(route/bilibili): user/followings & user/followers now require login

### PR DESCRIPTION
**Pull request by Qixingchen**
## 路由地址示例 / Example for the Proposed Route(s)

```routes
bilibili/user/followings/2267573/3
bilibili/user/followers/2267573/3
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [x] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

Since July 10, 2023, the relation/followers & relation/followings API requires login.   
It must use self-hosted server.  
The following content is the result of my local testing.  

<details><summary>user/followings:</summary>
<p>

``` XML
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<rss xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
<channel>
<title>
<![CDATA[ DIYgod 的 bilibili 关注 ]]>
</title>
<link>https://space.bilibili.com/2267573/#/fans/follow</link>
<atom:link href="http://127.0.0.1:1200/bilibili/user/followings/2267573/17875522?limit=2" rel="self" type="application/rss+xml"/>
<description>
<![CDATA[ DIYgod 的 bilibili 关注 - Made with love by RSSHub(https://github.com/DIYgod/RSSHub) ]]>
</description>
<generator>RSSHub</generator>
<webMaster>i@diygod.me (DIYgod)</webMaster>
<language>zh-cn</language>
<lastBuildDate>Sat, 15 Jul 2023 10:22:25 GMT</lastBuildDate>
<ttl>5</ttl>
<item>
<title>
<![CDATA[ DIYgod 新关注 林中小呜 ]]>
</title>
<description>
<![CDATA[ 林中小呜<br>新人UP主一枚，希望大家能够喜欢 <br>总计574 ]]>
</description>
<pubDate>Thu, 13 Jul 2023 09:17:09 GMT</pubDate>
<guid isPermaLink="false">https://space.bilibili.com/381396513</guid>
<link>https://space.bilibili.com/381396513</link>
</item>
<item>
<title>
<![CDATA[ DIYgod 新关注 秋葉aaaki ]]>
</title>
<description>
<![CDATA[ 秋葉aaaki<br>喂饭区UP主 / 爱发电：akibanzu / 纯个人<br>总计574 ]]>
</description>
<pubDate>Wed, 12 Jul 2023 15:44:31 GMT</pubDate>
<guid isPermaLink="false">https://space.bilibili.com/12566101</guid>
<link>https://space.bilibili.com/12566101</link>
</item>
</channel>
</rss>
```

</p>
</details>

<details><summary>user/followers:</summary>
<p>

``` XML
<rss xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
<channel>
<title>
<![CDATA[ DIYgod 的 bilibili 粉丝 ]]>
</title>
<link>https://space.bilibili.com/2267573/#/fans/fans</link>
<atom:link href="http://127.0.0.1:1200/bilibili/user/followers/2267573/17875522?limit=2" rel="self" type="application/rss+xml"/>
<description>
<![CDATA[ DIYgod 的 bilibili 粉丝 - Made with love by RSSHub(https://github.com/DIYgod/RSSHub) ]]>
</description>
<generator>RSSHub</generator>
<webMaster>i@diygod.me (DIYgod)</webMaster>
<language>zh-cn</language>
<lastBuildDate>Sat, 15 Jul 2023 10:20:08 GMT</lastBuildDate>
<ttl>5</ttl>
<item>
<title>
<![CDATA[ DIYgod 新粉丝 丶沉默是金灬 ]]>
</title>
<description>
<![CDATA[ 丶沉默是金灬<br><br>总计5332 ]]>
</description>
<pubDate>Fri, 14 Jul 2023 07:16:47 GMT</pubDate>
<guid isPermaLink="false">https://space.bilibili.com/354169244</guid>
<link>https://space.bilibili.com/354169244</link>
</item>
<item>
<title>
<![CDATA[ DIYgod 新粉丝 北国的雾 ]]>
</title>
<description>
<![CDATA[ 北国的雾<br>( ﾟ∀ﾟ)o彡゜えーりん！えーりん！<br>总计5332 ]]>
</description>
<pubDate>Thu, 13 Jul 2023 19:20:32 GMT</pubDate>
<guid isPermaLink="false">https://space.bilibili.com/82589501</guid>
<link>https://space.bilibili.com/82589501</link>
</item>
</channel>
</rss>
```

</p>
</details>

### CODE REVIEW

File: lib/v2/bilibili/followers.js [hunk 1]
```diff
@@ -1,11 +1,18 @@
 const got = require('@/utils/got');
 const cache = require('./cache');
+const config = require('@/config').value;
 
 module.exports = async (ctx) => {
     const uid = ctx.params.uid;
+    const login_uid = ctx.params.login_uid;
```
Comment (TonyRL) [inline lib/v2/bilibili/followers.js:7]: camelCase
docs: https://docs.rsshub.app/joinus/script-standard.html

File: lib/v2/bilibili/followers.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
 
     const name = await cache.getUsernameFromUID(ctx, uid);
 
+    const cookie = config.bilibili.cookies[login_uid];
+    if (cookie === undefined) {
+        throw Error('缺少对应 login_uid 的 Bilibili 用户登录后的 Cookie 值');
+    }
```
Comment (TonyRL) [inline lib/v2/bilibili/followers.js:14]: This check should be placed before declaring `uid` and `login_uid`.
You can add a link to the docs like below as well.
https://github.com/DIYgod/RSSHub/blob/e52c82971fec25a8dec8ec5652f00a8d0df28143/lib/v2/github/notifications.js#L8-L10

File: lib/v2/bilibili/followings.js [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
 module.exports = async (ctx) => {
     const uid = ctx.params.uid;
+    const login_uid = ctx.params.login_uid;
+
     const name = await cache.getUsernameFromUID(ctx, uid);
 
+    const cookie = config.bilibili.cookies[login_uid];
+    if (cookie === undefined) {
+        throw Error('缺少对应 login_uid 的 Bilibili 用户登录后的 Cookie 值');
+    }
```
Comment (TonyRL) [inline lib/v2/bilibili/followings.js:14]: Same as above.

File: lib/v2/bilibili/router.js [hunk 4]
```diff
@@ -31,8 +31,8 @@ module.exports = (router) => {
     router.get('/user/collection/:uid/:sid/:disableEmbed?', require('./userCollection'));
     router.get('/user/dynamic/:uid/:disableEmbed?', require('./dynamic'));
     router.get('/user/fav/:uid/:disableEmbed?', require('./userFav'));
-    router.get('/user/followers/:uid', require('./followers'));
-    router.get('/user/followings/:uid', require('./followings'));
+    router.get('/user/followers/:uid/:login_uid', require('./followers'));
+    router.get('/user/followings/:uid/:login_uid', require('./followings'));
```
Comment (TonyRL) [inline lib/v2/bilibili/router.js:35]: The same entries in `maintainer.js` also need to be updated.

File: docs/social-media.md [hunk 5]
```diff
@@ -70,11 +70,23 @@ Tiny Tiny RSS 会给所有 iframe 元素添加 `sandbox="allow-scripts"` 属性
...
 ### UP 主粉丝
 
-<Route author="Qixingchen" example="/bilibili/user/followers/2267573" path="/bilibili/user/followers/:uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到']" radar="1" rssbud="1"/>
+<Route author="Qixingchen" example="/bilibili/user/followers/2267573/3" path="/bilibili/user/followers/:uid/:login_uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到','用于登入的用户id,需要配置对应的 Cookie 值']" radar="1" rssbud="1" selfhost="1"/>
+
+::: warning 注意
+
+UP 主粉丝现在需要 b 站登录后的 Cookie 值，所以只能自建，详情见部署页面的配置模块。
+
+:::
```
Comment (TonyRL) [inline docs/social-media.md:79]:
```suggestion
<Route author="Qixingchen" example="/bilibili/user/followers/2267573/3" path="/bilibili/user/followers/:uid/:login_uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到','用于登入的用户id,需要配置对应的 Cookie 值']" radar="1" rssbud="1" selfhost="1">

::: warning 注意

UP 主粉丝现在需要 b 站登录后的 Cookie 值，所以只能自建，详情见部署页面的配置模块。

:::

</Route>
```

File: docs/social-media.md [hunk 6]
```diff
@@ -70,11 +70,23 @@ Tiny Tiny RSS 会给所有 iframe 元素添加 `sandbox="allow-scripts"` 属性
...
 ### UP 主关注用户
 
-<Route author="Qixingchen" example="/bilibili/user/followings/2267573" path="/bilibili/user/followings/:uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到']" radar="1" rssbud="1"/>
+<Route author="Qixingchen" example="/bilibili/user/followings/2267573/3" path="/bilibili/user/followings/:uid/:login_uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到','用于登入的用户id,需要配置对应的 Cookie 值']" radar="1" rssbud="1" selfhost="1"/>
+
+::: warning 注意
+
+UP 主关注用户现在需要 b 站登录后的 Cookie 值，所以只能自建，详情见部署页面的配置模块。
+
+:::
```
Comment (TonyRL) [inline docs/social-media.md:89]:
```suggestion
<Route author="Qixingchen" example="/bilibili/user/followings/2267573/3" path="/bilibili/user/followings/:uid/:login_uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到','用于登入的用户id,需要配置对应的 Cookie 值']" radar="1" rssbud="1" selfhost="1">

::: warning 注意

UP 主关注用户现在需要 b 站登录后的 Cookie 值，所以只能自建，详情见部署页面的配置模块。

:::

</Route>
```

**Comment by Qixingchen** [comment]
- Change to use camelCase
- Cookie config check move before declaring `uid`.And add a link to docs.But Cookie config need loginUid param.
- Update maintainer.js
- Fix docs/social-media.md and add description in docs/install/README.md

File: docs/social-media.md [hunk 7]
```diff
@@ -70,11 +70,27 @@ Tiny Tiny RSS 会给所有 iframe 元素添加 `sandbox="allow-scripts"` 属性
 
 ### UP 主粉丝
 
-<Route author="Qixingchen" example="/bilibili/user/followers/2267573" path="/bilibili/user/followers/:uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到']" radar="1" rssbud="1"/>
+<Route author="Qixingchen" example="/bilibili/user/followers/2267573/3" path="/bilibili/user/followers/:uid/:login_uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到','用于登入的用户id,需要配置对应的 Cookie 值']" radar="1" rssbud="1" selfhost="1">
```
Comment (TonyRL) [inline docs/social-media.md:73]:
```suggestion
<Route author="Qixingchen" example="/bilibili/user/followers/2267573/3" path="/bilibili/user/followers/:uid/:loginUid" :paramsDesc="['用户 id, 可在 UP 主主页中找到','用于登入的用户id,需要配置对应的 Cookie 值']" radar="1" rssbud="1" selfhost="1">
```

File: docs/social-media.md [hunk 8]
```diff
@@ -70,11 +70,27 @@ Tiny Tiny RSS 会给所有 iframe 元素添加 `sandbox="allow-scripts"` 属性
...
+UP 主粉丝现在需要 b 站登录后的 Cookie 值，所以只能自建，详情见部署页面的配置模块。
+
+:::
+
+</Route>
 
 ### UP 主关注用户
 
-<Route author="Qixingchen" example="/bilibili/user/followings/2267573" path="/bilibili/user/followings/:uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到']" radar="1" rssbud="1"/>
+<Route author="Qixingchen" example="/bilibili/user/followings/2267573/3" path="/bilibili/user/followings/:uid/:login_uid" :paramsDesc="['用户 id, 可在 UP 主主页中找到','用于登入的用户id,需要配置对应的 Cookie 值']" radar="1" rssbud="1" selfhost="1">
```
Comment (TonyRL) [inline docs/social-media.md:85]:
```suggestion
<Route author="Qixingchen" example="/bilibili/user/followings/2267573/3" path="/bilibili/user/followings/:uid/:loginUid" :paramsDesc="['用户 id, 可在 UP 主主页中找到','用于登入的用户id,需要配置对应的 Cookie 值']" radar="1" rssbud="1" selfhost="1">
```

**Pull request merged**